aip/auth/4120.md

```diff
@@ -0,0 +1,96 @@
+| Any other error | The authentication libraries **must not** fall back to the `"googleapis.com"` default value. The authentication libraries **should** do the same action as when the same error happens when retrieving the access token. |
+
+Additional recommendations:
+
+ - GCE credentials **should** retrieve the universe domain in a 'lazy' way, when the end-user requests the `universe_domain` via the property/get method. They **should not** do it during the credentials initialization.
+
+### Other requirements
+
+#### Authentication libraries **must not** use hardcoded non-constant information in the authentication flows.
+
+> This requirement applies **only** to the flows which are supported outside the GDU.
+
+The code for the authentication flows **must not** contain hardcoded endpoints, endpoint-alikes or other authentication information unless this information is constant across all flows for all universes. Any information that is possible to read from the credentials file should be read from the credentials file. For example, `scopes` contain the "googleapis.com" domain but are still valid outside the GDU because their values are consistent across all universes, whereas the STS endpoint "sts.googleapis.com" is not allowed because it is different for every universe.
+
+#### Authentication libraries **must** add the `universe_domain` property to the token authentication credentials constructor (if supported) and document the default behavior
+
+> This requirement applies **only** to the access token flow (if supported by the auth library).
+
+If the authentication library provides a "helper" credentials class that the end-user can initialize with the access token they already have, this class should additionally expose the `universe_domain` property in the constructor.
+The access tokens are not guaranteed to be end-user decodable or have universe domain information. Therefore the end-user should be able to conveniently specify which universe's access token they are providing. Preferably it should be in the same fashion that they provide the access token itself, so if e.g. the access token is a constructor parameter, the `universe_domain` should also be a parameter.
+This constructor property should have a default value of `"googleapis.com"` for backwards compatibility. This behavior should be documented in a way to draw the developer's attention since we want to prevent them from making a mistake and sending an access token from a different universe to the GDU "by default".
+
+### Authentication flows supported outside the GDU
+
+The official authentication flows which are supported outside the GDU are the following:
+
+ - Service Account Credentials (through use of Service Account JWT Access Credentials only)
```

Comment:
Nit: service account key credential json files

Nit: use of Self-Signed JWT Access Tokens (and ID tokens?)

Reply:
In code, we call these credentials "Service Account Credentials", so I think the name here is appropriate.

Reply:
Changed to `Service Account Key Credentials`